### DISCUSSION TITLE
App Store Notification Subtype

### DISCUSSION THREAD

**Discussion by kiritokatklian** [imdhemy/laravel-in-app-purchases#148] [General]
Hey! First of all great work on v1 💪

I was wondering if subtype notifications are supported for App Store. For example the `Subscribe` notification has subtypes of `INITIAL_BUY` and `RESUBSCRIBE`.

I found a couple references to subtype, so I'm assuming it's already in the package. However, I couldn't find a method to access the subtypes from an event. I might be looking in the wrong place.

Here is the list of available subtypes: https://developer.apple.com/documentation/appstoreservernotifications/subtype

**Comment by imdhemy** [comment]
@kiritokatklian 

Thank you for your nice words.

I will create an issue from this discussion as we missed this method.

**Reply by rubanraj7** [reply]
Any change this is complete?

**Reply by imdhemy** [reply]
It's already supported. 

https://github.com/imdhemy/laravel-in-app-purchases/blob/ff47398cdf29be71bc8f6a632536fcbfa5fbfa99/src/ServerNotifications/AppStoreV2ServerNotification.php#L65-L68
